src/main/ngapp/src/app/auth/auth.service.ts

```diff
@@ -17,25 +19,26 @@ export class AuthService {
   private eventAuthError = new BehaviorSubject<string>("");
   eventAuthError$ = this.eventAuthError.asObservable();
   newUser: any;
-
   user: User;
 
-  constructor(public afAuth: AngularFireAuth, public router: Router) {
-    this.afAuth.authState.subscribe((user) => {
-      if (user) {
-        this.user = user;
-        localStorage.setItem('user', JSON.stringify(this.user));
-      } else {
-        localStorage.setItem('user', null);
-      }
-    });
+  constructor(public afAuth: AngularFireAuth, private router: Router, private http: HttpClient) {
+    // this.afAuth.authState.subscribe((user) => {
+    //   if (user) {
+    //     this.user = user;
+    //     localStorage.setItem('user', JSON.stringify(this.user));
+    //   } else {
+    //     localStorage.setItem('user', null);
+    //   }
+    // });
```

Comment:
why was this deleted? shouldnt we be checking the authentification state?

Reply:
I decided to manually add the auth state to localStorage whenever a user logsin/signs up/ registers

Reply:
so the authState just lasts for the duration that the application tab/window is open and disappears when it is closed?

Reply:
never mind. i think i got it